bolt-sidecar/src/state/execution.rs

```diff
@@ -93,6 +99,7 @@ impl<C: StateFetcher> ExecutionState<C> {
             slot: 0,
             account_states: HashMap::new(),
             block_templates: HashMap::new(),
+            chain_id: client.get_chain_id().await?,
             client,
         })
```

Comment:
Another way to do this can be to pass the chain ID value from config as a parameter while initializing the execution state, which felt to me as a dirty/ un-clean way.